tools/clang_tidy/lib/clang_tidy.dart

```diff
@@ -165,20 +165,21 @@ class ClangTidy {
   /// Given a build commands json file, and the files with local changes,
   /// compute the lint commands to run.
   @visibleForTesting
-  List<Command> getLintCommandsForChangedFiles(
+  Future<List<Command>> getLintCommandsForChangedFiles(
     List<dynamic> buildCommandsData,
     List<io.File> changedFiles,
-  ) {
-    final List<Command> buildCommands = <Command>[
-      for (final dynamic c in buildCommandsData)
-        Command.fromMap(c as Map<String, dynamic>),
-    ];
-
-    return <Command>[
-      for (final Command c in buildCommands)
-        if (c.containsAny(changedFiles))
-          c,
-    ];
+  ) async {
+    final List<Command> buildCommands = <Command>[];
+    for (final dynamic data in buildCommandsData) {
+      final Command command = Command.fromMap(data as Map<String, dynamic>);
+      final LintAction lintAction = await command.lintAction;
+      if (lintAction != LintAction.skipMissing &&
```

Comment:
Skip files that don't exist in case there is anything stale in `host_debug/compile_commands.json`.